test/testpy/__init__.py

```diff
@@ -64,7 +65,22 @@ def GetCommand(self):
       # PORT should match the definition in test/common.js.
       env = { 'PORT': int(os.getenv('NODE_COMMON_PORT', '12346')) }
       env['PORT'] += self.thread_id * 100
-      result += flags_match.group(1).strip().format(**env).split()
+      flag = flags_match.group(1).strip().format(**env).split()
+      # The following block reads config.gypi to extract the v8_enable_inspector
+      # value. This is done to check if the inspector is disabled in which case
+      # the '--inspect' flag cannot be passed to the node process as it will
+      # that will cause node to exit and report the test as failed. The use case
+      # is currently when Node is configured --without-ssl and the tests should
+      # still be runnable but skip any tests that require ssl (which includes the
+      # inspector related tests).
+      with open('config.gypi', 'r') as f:
+        s = f.read()
+        config_gypi = ast.literal_eval(s)
+        v8_disable_inspector = config_gypi['variables']['v8_enable_inspector'] == 0
+        if flag[0].startswith('--inspect') and v8_disable_inspector:
+          print('Skipping as inspector is disabled')
+        else:
+          result += flag
```

Comment:
Does this run once per run or once per test?  The latter would be a little wasteful.